test_unstructured/partition/test_auto.py

```diff
@@ -187,7 +188,17 @@ def test_auto_partition_doc_with_file(mock_docx_document, expected_docx_elements
         PartitionStrategy.OCR_ONLY,
     ],
 )
-def test_partition_forwards_strategy_arg_to_partition_docx(request: FixtureRequest, strategy: str):
+def test_partition_forwards_strategy_arg_to_partition_docx_and_its_brokers(
```

Comment:
Is there a test we can add that tests this for all brokering partitioners? So if something isn't passed we're able to catch it.

Reply:
Maybe, but I think the solution is not being selective about what parameters we pass along from `partition()` to delegate partitioners. I go into more details in the write-up I sent, but there shouldn't be a good reason for us to be explicitly calling out which parameters get forwarded. We can do something like:
```python
        elements = _partition_doc(filename, file=file, **kwargs)
```
instead of what we're doing now which looks like this:
```python
        elements = _partition_doc(
            filename=filename,
            file=file,
            infer_table_structure=infer_table_structure,
            languages=languages,
            detect_language_per_element=detect_language_per_element,
            starting_page_number=starting_page_number,
            strategy=strategy,
            **kwargs,
        )
```

There's some work to get there, which I went into in my write-up, but the result is basically maintenance-free.

All partitioners get all arguments from the call, they pick out which ones they want and ignore the rest.

Any new parameter that a partitioner starts using will already be in its `kwargs`. All it has to do is create a parameter definition for that one and it will get populated on each call.